### PR TITLE
add file ShellSortExample.swift

### PR DESCRIPTION
**Pull request by Moon1102**
update code.

### CODE REVIEW

**Comment by kelvinlauKL** [comment]
I'm not convinced this is an improvement over the current implementation. `shellsort.swift` looks succinct already, and follows the current explanation of it being an improvement over insertion sort. 

What sort of improvement do you want to show with your example (compared to the current implementation)?

**Comment by Moon1102** [comment]
I agree with you there.
But I think, in the same way, my algorithm takes less time. Maybe my advantage is here.

part1
part2

File: Shell Sort/ShellSortExample.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+//  ShellSortExample.swift
+//  
+//
+//  Created by Cheer on 2017/2/26.
+//
+//
+
+import Foundation
+
+public func shellSort(_ list : inout [Int])
```
Comment (kelvinlauKL) [inline Shell Sort/ShellSortExample.swift]: Make sure the `:` hugs the parameter name.

File: Shell Sort/ShellSortExample.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+{
+    var sublistCount = list.count / 2
+    
+    while sublistCount > 0
```
Comment (kelvinlauKL) [inline Shell Sort/ShellSortExample.swift]: For consistency, let's have the braces trail the `while` statement.

File: Shell Sort/ShellSortExample.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        for index in 0..<list.count{
```
Comment (kelvinlauKL) [inline Shell Sort/ShellSortExample.swift]: Add a space between `list.count` and the `{`.

Comment (Moon1102) [inline Shell Sort/ShellSortExample.swift]: Sure. I'll change my code to suit style guide. Thanks for your direction.

**Comment by mitghi** [comment]
@Moon1102 Your implementation is much better, as it eliminated the function call and iterator overheads.

**Comment by kelvinlauKL** [comment]
Looks good. Thanks @Moon1102 @mitghi !

**Comment by Moon1102** [comment]
Thanks @kelvinlauKL  @mitghi

**Comment by daisysomus** [comment]
@Moon1102 Your implementation has bugs. If you try this array: [64, 20, 50, 21, 72, 10, 23, -1, 4], you won't get the right answer.

**Comment by Moon1102** [comment]
@daisysomus  Thank you for pointing out the mistake.

**Comment by Moon1102** [comment]
image

**Pull request merged**